wgpu-core/src/resource.rs

```diff
@@ -215,8 +215,8 @@ pub struct Texture<B: hal::Backend> {
 
 #[derive(Clone, Debug, Error)]
 pub enum TextureDimensionError {
-    #[error("too many layers ({0}) for texture array")]
-    TooManyLayers(u32),
+    #[error("one of the dimensions is zero")]
+    Zero,
```

Comment:
Could use [`TextureErrorDimension`](https://github.com/gfx-rs/wgpu/blob/master/wgpu-core/src/command/transfer.rs#L36) to make this error better.

Reply:
And if you want to get real fancy, to detect this, you could do

```rust
match (width, height, depth) {
    (0, _, _) => return Err(Zero(TextureErrorDimension::X)),
    (_, 0, _) => return Err(Zero(TextureErrorDimension::Y)),
    (_, _, 0) => return Err(Zero(TextureErrorDimension::Z)),
    _ => {}
}
```